pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -462,6 +467,74 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileBuildRun) updateBuildRunUsingTaskRunCondition(ctx context.Context, buildRun *buildv1alpha1.BuildRun, taskRun *v1beta1.TaskRun, trCondition *apis.Condition) error {
+	var reason, message string = trCondition.Reason, trCondition.Message
+
+	switch v1beta1.TaskRunReason(reason) {
+	case v1beta1.TaskRunReasonTimedOut:
+		reason = "BuildRunTimeout"
+		message = fmt.Sprintf("BuildRun %s failed to finish within %s",
+			taskRun.GetLabels()[buildv1alpha1.LabelBuildRun],
```

Comment:
nit - we probably don't need the name since in the message will be inside the BuildRun object.

Reply:
From a user that is looking at the Conditions directly from the `BuildRun` I agree, but for the scenarios where someone is building a workflow on top(e.g. Shipwright/CLI), this will help to make the message very explicit. Therefore I would opt to not change this.